bms_rangecheck.c

```diff
@@ -0,0 +1,118 @@
+/* *************************************************************************
+* File Name   :	bms_rangecheck.c
+* Description : Battery Management System(BMS)
+* Functions	  :
+* ************************************************************************* */
+
+/* ***************************** Header Files ***************************** */
+#include "stdio.h"
+#include "assert.h"
+#include "bms_rangecheck.h"
+
+/* *************************************************************************
+* Function Name : rangeCheckTemp
+* Description   : To check the temperature range
+* Arguments	    : Temperature value
+* Returns		: (0,1) Within range or not
+* ************************************************************************* */
+int rangeCheckTemp(float temperature) {
+	
+	if(TEMP_UNIT == TEMP_FAHRENHEIT)
+	{
+		temperature = (temperature - 32) * (5/9);
+	}
+	else
+	{
+	}
+	
+    if(temperature < MIN_TEMP)

```

Comment:
This function has CCN > 3 and the < and > checks repeat across the next functions too.
Remove the duplication with a range check and the complexity will come down on its own.